pandas/io/json/_json.py

```diff
@@ -315,19 +335,20 @@ def _write(
         date_unit,
         iso_dates,
         default_handler,
+        indent,
     ):
-        data = super()._write(
-            obj,
+        table_obj = OrderedDict((("schema", self.schema), ("data", obj)))
+        serialized = super()._write(
+            table_obj,
             orient,
             double_precision,
             ensure_ascii,
             date_unit,
             iso_dates,
             default_handler,
+            indent,
         )
-        serialized = '{{"schema": {schema}, "data": {data}}}'.format(
```

Comment:
Previously this was creating a formatted string from individual dumps of the JSON data. To get indent to work with that formatted string adds a lot of code, so instead I put everything into an OrderedDict and am letting the JSON encoder do the heavy lifting.

This fixed a bug as indicated in the whatsnew but also shed some light on #28256 which I think is best tackled separately.

**Note** that this also changes the whitespace in use for the table format, which was inconsistent before anyway. The JSON encoder doesn't put a space after each colon whereas this formatted string did. Whitespace is insignificant in JSON so I don't think this really needs a whatsnew mention but can be convinced otherwise